sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -349,10 +349,18 @@ object ScalaReflection extends ScalaReflection {
         // TODO: add walked type path for map
         val TypeRef(_, _, Seq(keyType, valueType)) = t
 
+        val classNameForKey = getClassNameFromType(keyType)
+        val classNameForValue = getClassNameFromType(valueType)
+
+        val newTypePath =
+          s"""- map key class: "$classNameForKey",
+             | value class: "$classNameForValue"""".stripMargin +:
+          walkedTypePath
+
```

Comment:
Is this change related to this pr?

Reply:
Not strictly, but consistent between `ScalaReflection.deserializeFor` and `JavaTypeInference.deserializeFor`. It is marked as TODO and symmetric part is addressed in `JavaTypeInference.deserializeFor` so why not address it to `ScalaReflection.deserializeFor` too? 

If we would like to address this from out of this PR I'll add follow-up PR against this.

Reply:
Can you add tests for this code path in `ScalaReflectionSuite`?

Reply:
Adding a UT just for testing path addition looks overkill. If you're unsure about this chance unless I add a new test for this, I'd rather roll back (adding TODO back) and address this in follow-up PR which may add a new object tracking walkedTypePath for both Java/Scala.